File: src/app/models/ApiResponse.ts

export class APIResponse<T> {
    success: boolean;
    message: string;
    data: T;
    errors: string[];

    constructor(success: boolean, message: string, data: T, errors: string[] = []) {
        this.success = success;
        this.message = message;
        this.data = data;
        this.errors = errors;
    }    
    
}

File: src/main/java/com/onlinevoting/dto/ApiResponse.java
package com.onlinevoting.dto;

import java.util.List;

import lombok.Getter;
import lombok.Setter;


@Getter
@Setter
public class ApiResponse<T> {
    private boolean success;
    private T data;
    private List<String> errors;
    private String message;

    public ApiResponse(){

    }

    public ApiResponse(boolean success, T data, List<String> errors) {
        this.success = success;
        this.data = data;
        this.errors = errors;
    }


    public ApiResponse(boolean success, T data, List<String> errors, String message) {
        this.success = success;
        this.data = data;
        this.errors = errors;
        this.message = message;
    }

}
